pkg/daemon/controller/node.go

```diff
@@ -228,10 +228,34 @@ func (c *Controller) reconcileNodeInfo() error {
 		return fmt.Errorf("list address for all interfaces failed: %v", err)
 	}
 
+	var nodeLocalVxlanAddr []netlink.Addr
+	for _, addr := range existAllAddrList {
+		// Add vtep ip and node object ip by default.
+		isNodeObjectAddr := false
+		for _, nodeObjectAddr := range thisNode.Status.Addresses {
+			if nodeObjectAddr.Address == addr.IP.String() {
+				isNodeObjectAddr = true
+				break
+			}
+		}
+
+		if isNodeObjectAddr {
+			nodeLocalVxlanAddr = append(nodeLocalVxlanAddr, addr)
+			continue
+		}
+
+		// Add extra node local vxlan ip.
+		for _, cidr := range c.config.ExtraNodeLocalVxlanIPCidrs {
```

Comment:
Two questions,
1. why include ip but not exclude it? IMO, some invalid ip addresses should be ignored.
2. why multiple IP addresses will be pick up for vxlan devices, vtep seems to need only one?

Reply:
IP addresses for the vxlan devices are not only used to forward packets, which make node as a vtep gateway, but also to be used as the source addresses when node need to access pod.

Reply:
I don't think this is a good feature for us now, if pod want to access node, only the "main" IP will and should be chosen, even the node may have several IPs, IMO, these should be treated as external IPs.

Reply:
BTW, do you have some user cases about this?

Reply:
> I don't think this is a good feature for us now, if pod want to access node, only the "main" IP will and should be chosen, even the node may have several IPs, IMO, these should be treated as external IPs.

This is exactly this pr are doing. Before this pr, every node will get all of local ip addresses on it to assigned to vxlan interface. After this pr, only the "main" IP will and should be chosen.

> BTW, do you have some user cases about this?

Choosing extra node local ip addresses as local vxlan address is for extra usages, in case someone want to use it.

Reply:
Gotcha!